applications/binaryfile-recorder/src/main/java/com/paritytrading/nassau/binaryfile/recorder/Recorder.java

```diff
@@ -170,46 +165,44 @@ public void endOfSession(MoldUDP64Client session) {
     }
 
     private static void receive(MoldUDP64Client client) throws IOException {
-        Selector selector = Selector.open();
+        try (Selector selector = Selector.open()) {
 
-        SelectionKey key = client.getChannel().register(selector, SelectionKey.OP_READ);
+            SelectionKey key = client.getChannel().register(selector, SelectionKey.OP_READ);
 
-        SelectionKey requestKey = client.getRequestChannel().register(selector, SelectionKey.OP_READ);
+            SelectionKey requestKey = client.getRequestChannel().register(selector, SelectionKey.OP_READ);
 
-        while (receive) {
-            int numKeys = selector.select();
-            if (numKeys > 0) {
-                if (selector.selectedKeys().contains(key))
-                    client.receive();
+            while (receive) {
+                int numKeys = selector.select();
+                if (numKeys > 0) {
+                    if (selector.selectedKeys().contains(key))
+                        client.receive();
 
-                if (selector.selectedKeys().contains(requestKey))
-                    client.receiveResponse();
+                    if (selector.selectedKeys().contains(requestKey))
+                        client.receiveResponse();
 
-                selector.selectedKeys().clear();
+                    selector.selectedKeys().clear();
+                }
             }
-        };
-
-        selector.close();
+        }
     }
 
     private static void receive(SoupBinTCPClient client) throws IOException {
-        Selector selector = Selector.open();
+        try (Selector selector = Selector.open()) {
 
-        SelectionKey key = client.getChannel().register(selector, SelectionKey.OP_READ);
```

Comment:
I removed the unused local `SelectionKey key` here which probably should have been in its own PR.

Reply:
I think it's fine to clean up here. Nice find! 👍